lib/optionInterfaces.ts

```diff
@@ -49,3 +63,29 @@ export interface CallOption {
 	 */
 	sdpTransform?: Function;
 }
+
+
+type SerializationTypeConstructor = new (
+	peerId: string,
+	provider: Peer,
+	options: any,
+) => DataConnection;
+
+/**
+ * A mapping of serialization types to their implementations.
+ * Should map 1:1 with the serialization type property in the class.
+ */
+export type SerializerMapping = {
+	[key in SerializationType]: SerializationTypeConstructor;
```

Comment:
One concern id raise over doing this is that it locks the user into ONLY the SerializerTypes that we provide, Changing this back to 
```ts
[key: string]: new (
		peerId: string,
		provider: Peer,
		options: any,
	) => DataConnection;
```
reverts this